pkg/controller/history/controller_history.go

```diff
@@ -454,3 +452,31 @@ func (fh *fakeHistory) ReleaseControllerRevision(parent metav1.Object, revision
 	}
 	return clone, fh.indexer.Update(clone)
 }
+
+func getPodLabelsFromData(data runtime.RawExtension) map[string]string {
+	var dat map[string]interface{}
+	labelMap := make(map[string]string)
+	if err := json.Unmarshal(data.Raw, &dat); err != nil {
+		return labelMap
```

Comment:
If the raw data does not parse we should not be silently swallowing that error.

Reply:
Will a warning in log work? I didn't want it to break like previously if the problem is only with parsing of labels, as they aren't being used anywhere anyway.